src/types.rs

```diff
@@ -487,6 +487,39 @@ impl<T> Line<T>
 #[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub struct LineString<T>(pub Vec<Point<T>>) where T: Float;
 
+impl<T: Float> LineString<T> {
+    /// Return an `Line` iterator that yields one `Line` for each line segment
+    /// in the `LineString`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::{Line, LineString, Point};
+    ///
+    /// let mut points = vec![(0., 0.), (5., 0.), (7., 9.)];
+    /// let linestring: LineString<f32> = points.into_iter().collect();
+    ///
+    /// let mut lines = linestring.lines();
+    /// assert_eq!(
+    ///     Some(Line::new(Point::new(0., 0.), Point::new(5., 0.))),
+    ///     lines.next()
+    /// );
+    /// assert_eq!(
+    ///     Some(Line::new(Point::new(5., 0.), Point::new(7., 9.))),
+    ///     lines.next()
+    /// );
+    /// assert!(lines.next().is_none());
+    /// ```
+    pub fn lines<'a>(&'a self) -> Box<Iterator<Item = Line<T>> + 'a> {
```

Comment:
we can get rid of this `Box` once we can do `-> impl Iterator<...>` in stable rust